googleapi/googleapi.go

```diff
@@ -211,6 +211,20 @@ type MediaOption interface {
 	setOptions(o *MediaOptions)
 }
 
+type bufferOption []byte
+
+func (bp bufferOption) setOptions(o *MediaOptions) {
+	o.Buffer = bp
+}
+
+// WithBuffer returns MediaOption which sets buffer used for media uploads.
+// It needs to be at least MinUploadChunkSize long.
```

Comment:
This is never validated. I think if the buffer is less than this it should not be used and an allocation occurs.(As it would today)

Reply:
Added.